src/libbson/tests/json/bson_corpus/top.json

```diff
@@ -151,6 +166,10 @@
             "description": "Bad $code (type is number, not string)",
             "string": "{\"a\" : {\"$code\" : 42}}"
         },
+        {
+            "description": "Bad $code (type is number, not string) when $scope is also present",
```

Comment:
Related to https://github.com/mongodb/specifications/commit/d923c3af2b59fbd19d473a895930c3b20707ec41 and [DRIVERS-1326](https://jira.mongodb.org/browse/DRIVERS-1326). This will let us resolve [CDRIVER-3742](https://jira.mongodb.org/browse/CDRIVER-3742).